AbilityChanger/Abilities/DreamNail.cs

```diff
@@ -0,0 +1,170 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace AbilityChanger
+{
+    public class DreamNail : AbilityManager, IStartable, IChargable, ICancellable, ITriggerable, IContacting, ICompletable,IOngoing
+    {
+        public override string abilityName { get; protected set; } = Abilities.DREAMNAIL;
+        public override bool hasDefaultAbility() => PlayerDataPatcher.GetBoolInternal(PlayerDataPatcher.hasDreamNail);
+        public override string inventoryTitleKey { get; protected set; } = "INV_NAME_DREAMNAIL";
+        public override string inventoryDescKey { get; protected set; } = "INV_DESC_DREAMNAIL";
+        public override GameObject getIconGo() => InvGo.Find("Dream Nail");
+        public DreamNail() : base()
+        {
+            On.EnemyDreamnailReaction.RecieveDreamImpact += Contact;
+        }
+
+        private void Contact(On.EnemyDreamnailReaction.orig_RecieveDreamImpact orig, EnemyDreamnailReaction self)
+        {
+            orig(self);
+            HandleContact(self.gameObject);
```

Comment:
i think contact is the only one i want to happen even when orig happens, rest i want to ideally replace.